cmd/juju/machine/upgradeseries.go

```diff
@@ -243,6 +246,41 @@ func (c *upgradeSeriesCommand) UpgradeSeriesPrepare(ctx *cmd.Context) error {
 	return nil
 }
 
+func (c *upgradeSeriesCommand) promptConfirmation(ctx *cmd.Context) ([]string, error) {
+	affectedUnits, err := c.upgradeMachineSeriesClient.UpgradeSeriesValidate(c.machineNumber, c.series)
```

Comment:
If we need the units elsewhere we should extract the validation method out of this method and pass them in rather than returning them. I may not be seeing something though, let me know.

Reply:
We don't really need the units elsewhere; we need the applications. In the prepare case, I am deriving the applications from the units to save an API round-trip.

On the complete side, we get them through the new `Aplications` API method.

We could get the units separately, but that would need a new API call too. This looks like the lightest touch.